flight/Modules/Actuator/triflight.c

```diff
@@ -0,0 +1,1018 @@
+
+	float angle_diff = servo_setpoint_angle - triflightStatus->ServoAngle;
+
+	if (fabsf(angle_diff) > max_phase_shift)
+	{
+		angle_diff = max_phase_shift * angle_diff / fabsf(angle_diff);
+	}
+
+	float future_servo_angle = bound_min_max(triflightStatus->ServoAngle + angle_diff,
+	                                         TRI_TAIL_SERVO_ANGLE_MID - triflightSettings->ServoMaxAngle,
+	                                         TRI_TAIL_SERVO_ANGLE_MID + triflightSettings->ServoMaxAngle);
+
+	float correction = (throttle * getPitchCorrectionAtTailAngle(DEG2RAD * future_servo_angle, triflightSettings->MotorThrustFactor)) - throttle;
+
+	return correction;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////
+
+#if !defined(ARCH_POSIX) && !defined(ARCH_WIN32)
+
+#include "gyros.h"
+#include "manualcontrolcommand.h"
+
+#define TAIL_THRUST_FACTOR_MIN  (1.0f)
+#define TAIL_THRUST_FACTOR_MAX  (40.0f)
+
+typedef enum {
+	TT_IDLE = 0,
+	TT_WAIT,
+	TT_ACTIVE,
+	TT_WAIT_FOR_DISARM,
+	TT_DONE,
+	TT_FAIL,
+} tailTuneState_e;
+
+typedef enum {
+	TT_MODE_NONE = 0,
+	TT_MODE_THRUST_TORQUE,
+	TT_MODE_SERVO_SETUP,
+} tailtuneMode_e;
+
+typedef enum {
+	SS_IDLE = 0,
+	SS_CALIB,
+} servoSetupState_e;
+
+typedef enum {
+	SS_C_IDLE = 0,
+	SS_C_CALIB_MIN_MID_MAX,
+	SS_C_CALIB_SPEED,
+} servoSetupCalibState_e;
+
+typedef enum {
+	SS_C_MIN = 0,
+	SS_C_MID,
+	SS_C_MAX,
+} servoSetupCalibSubState_e;
+
+typedef struct tailTune_s {
+	tailtuneMode_e mode;
+
+	struct thrustTorque_t
+	{
+		tailTuneState_e state;
+
+		uint32_t startBeepDelay_us;
+		uint32_t timestamp_us;
+
+		uint32_t lastAdjTime_us;
+		struct servoAvgAngle_t
+		{
+			float sum;
+			float num_of;
+		} servoAvgAngle;
+		}tt;
+
+	struct servoSetup_t
+{
+		servoSetupState_e state;
+
+		float servoVal;
+
+		struct servoCalib_t
+		{
+			bool done;
+			bool waitingServoToStop;
+
+			servoSetupCalibState_e state;
+			servoSetupCalibSubState_e subState;
+
+			uint32_t timestamp_us;
+
+			struct average_t
+			{
+				float result;
+				float sum;
+				float num_of;
+			} avg;
+		}cal;
+	}ss;
+} tailTune_t;
+
+static tailTune_t tailTune = {.mode = TT_MODE_NONE};
+
+typedef enum {
+	BUZZER_OFF = 0,
+	BUZZER_ON,
+} buzzerState_e;
+
+typedef struct buzzer_s {
+	buzzerState_e state;
+	uint32_t duration;
+} buzzer_t;
+
+static buzzer_t buzzer = {.state = BUZZER_OFF, .duration = 0};
+
+/**
+ * @}
+ * @}
+ */
+static void buzzerControl(ActuatorSettingsData  *actuatorSettings,
+                          TriflightStatusData   *triflightStatus)
+{
+	static uint32_t buzzer_timestamp;
+	static uint32_t duration;
+
+	if (triflightStatus->BuzzerChannel == 255)
+		return;
+
+	if (buzzer.state == BUZZER_ON)
+	{
+		if (PIOS_DELAY_GetuSSince(buzzer_timestamp) >= duration)
```

Comment:
a separate set of buzzer infrastructure for this isn't great-- we need to figure out a way to unify all the buzzer things.